### PR TITLE
gh-947: split `positions_from_delta` into internal functions

### PR DESCRIPTION
**Pull request by paddyroddy**
# Description

Split up the `positions_from_delta` function. The function is doing too much and is cumbersome to test. We've also noticed that it regularly fails the benchmarking. This will take some work to get to something we are happy with. This first step creates some internal functions that can be individually tested.

Note these functions aren't meant to be perfect and come from the suggestions in #413.

Closes: #947

## Checks

- [x] Is your code passing linting?
- [x] Is your code passing tests?
- [x] Have you added additional tests (if required)?
- [ ] Have you modified/extended the documentation (if required)?
- [ ] Have you added a one-liner changelog entry above (if required)?

### CODE REVIEW

File: tests/core/test_points.py [hunk 1]
```diff
@@ -101,6 +104,145 @@ def test_loglinear_bias(
...
+
+
+def test_sample_number_galaxies(
+    compare: type[Compare],
+    urng: UnifiedGenerator,
+    xp: ModuleType,
+) -> None:
+    n_in = xp.repeat(xp.asarray([0.0, 24751.77674965]), 6)
+
+    n = glass.points._sample_number_galaxies(n_in, rng=urng)
```
[the remark this answers is not in the document]
Comment (connoraird) [reply tests/core/test_points.py]: I meant add an additional test where `rng` is None
